recirq/quantum_chess/quantum_board.py

```diff
@@ -255,13 +258,19 @@ def sample_with_ancilla(self, num_samples: int
                         f'Discarded {error_count} from error mitigation '
                         f'{noise_count} from noise and '
                         f'{post_count} from post-selection\n')
+                    t1 = time.perf_counter()
+                    self.debug_log += (f"sample_with_ancilla takes {t1 - t0:0.4f} seconds.\n")
+                    self.timing_stats.append(t1 - t0)
                     return (rtn, ancilla)
         else:
             rtn = [self.state] * num_samples
             self.debug_log += (
                 f'Discarded {error_count} from error mitigation '
                 f'{noise_count} from noise and {post_count} from post-selection\n'
             )
+        t1 = time.perf_counter()
+        self.debug_log += (f"sample_with_ancilla takes {t1 - t0:0.4f} seconds.\n")
+        self.timing_stats.append(t1 - t0)
```

Comment:
Let's add a short test for this functionality as well.

Reply:
Thanks Doug. What is the right way to run tests? I've tried to add the following lines to quantum_board_test.py:

```
def test_record_time():
    b = qb.CirqBoard(u.squares_to_bitboard(['b8', 'b5', 'c7']))
    assert b.perform_moves(
        'b8a6c6:SPLIT_JUMP:BASIC',
        'b5a6:PAWN_CAPTURE:BASIC',
        'c6b8:JUMP:BASIC',
        'c7c5:PAWN_TWO_STEP:BASIC',
        'b5c6:PAWN_EP:BASIC',
    )
    b.print_timing_stats()

def main():
    test_record_time()

if __name__ == "__main__":
    main()
```

But it returns the error:
AttributeError: 'CirqBoard' object has no attribute 'print_timing_stats'

Reply:
Don't add the main() function to the test.  You can just add test_record_time (or any function that begins with "test_" and then run "pytest path/to/quantum_board_test.py".  

I am not sure why you are getting the error.  Are you sure you are on the correct branch?

I would also assert that "sample_with_ancilla takes" and "seconds" are in the debug log and that there is a positive entry in the timing_stats.  Potentially, you could mock time.perf_counter in order to get an exact value for the time, but you probably don't need to do that for this example.

Reply:
Thanks Doug. The test is updated. I did not know about "pytest". Thanks for letting me know how to test the test file.